eclair-core/src/main/scala/fr/acinq/eclair/io/PeerReadyNotifier.scala

```diff
@@ -68,13 +70,28 @@ object PeerReadyNotifier {
               // polling the switchboard. This makes more sense for long timeouts such as the ones used for async payments.
               context.system.eventStream ! EventStream.Subscribe(context.messageAdapter[PeerConnected](e => SomePeerConnected(e.nodeId)))
               context.system.eventStream ! EventStream.Subscribe(context.messageAdapter[PeerDisconnected](e => SomePeerDisconnected(e.nodeId)))
-              waitForPeerConnected(replyTo, remoteNodeId, switchboard, context, timers)
+              findSwitchboard(replyTo, remoteNodeId, context, timers)
           }
         }
       }
     }
   }
 
+  private def findSwitchboard(replyTo: ActorRef[Result], remoteNodeId: PublicKey, context: ActorContext[Command], timers: TimerScheduler[Command]): Behavior[Command] = {
+    context.system.receptionist ! Receptionist.Find(Switchboard.SwitchboardServiceKey, context.messageAdapter[Receptionist.Listing](WrappedListing))
+    Behaviors.receiveMessagePartial {
+      case WrappedListing(Switchboard.SwitchboardServiceKey.Listing(listings)) =>
+        listings.headOption match {
+          case Some(switchboard) =>
+              waitForPeerConnected(replyTo, remoteNodeId, switchboard, context, timers)
+          case None =>
+            context.log.error("no switchboard found")
+            replyTo ! PeerUnavailable(remoteNodeId)
+            Behaviors.stopped
```

Comment:
That's not very robust, shouldn't we just retry after e.g. 500ms? Isn't that the general way we should use the receptionist (poll it until available)?